src/config-schema.ts

```diff
@@ -1,60 +1,88 @@
-import { Type } from '@openmrs/esm-framework';
+import { Type, validators } from '@openmrs/esm-framework';
 import { type RenderType } from '@openmrs/esm-form-engine-lib';
 import type { QuestionType } from '@types';
 
+const allowedQuestionTypes: Array<QuestionType> = [
+  'control',
+  'encounterDatetime',
+  'encounterLocation',
+  'encounterProvider',
+  'encounterRole',
+  'obs',
+  'obsGroup',
+  'patientIdentifier',
+  'personAttribute',
+  'testOrder',
+  'programState',
+];
+
+const allowedFieldTypes: Array<RenderType> = [
+  'checkbox',
+  'checkbox-searchable',
+  'content-switcher',
+  'date',
+  'datetime',
+  'drug',
+  'encounter-location',
+  'encounter-provider',
+  'encounter-role',
+  'fixed-value',
+  'file',
+  'group',
+  'number',
+  'problem',
+  'radio',
+  'repeating',
+  'select',
+  'text',
+  'textarea',
+  'toggle',
+  'ui-select-extended',
+  'workspace-launcher',
+  'markdown',
+  'extension-widget',
+  'select-concept-answers',
+];
+
 export const configSchema = {
   questionTypes: {
     _type: Type.Array,
+    _elements: {
+      _type: Type.String,
+    },
     _description: 'Provides information that the processor uses to render a field',
-    _default: [
-      'control',
-      'encounterDatetime',
-      'encounterLocation',
-      'encounterProvider',
-      'encounterRole',
-      'obs',
-      'obsGroup',
-      'patientIdentifier',
-      'personAttribute',
-      'testOrder',
-      'programState',
-    ],
+    _default: allowedQuestionTypes,
```

Comment:
I am unsure how the authors of esm-form-builder intended for this to be used. But you don't need to worry about that :)